appeals/web/src/server/lib/mappers/data/appeal/mapper.js

```diff
@@ -78,25 +82,35 @@ export async function initialiseAndMapAppealData(
 
 	const userHasUpdateCasePermission = userHasPermission(permissionNames.updateCase, session);
 
-	/** @type {Record<string, SubMapper>} */
-	const submappers = submaps[appealDetails.appealType];
-
-	/** @type {{appeal: MappedInstructions}} */
-	const mappedData = {
-		appeal: {}
+	return {
+		appealDetails,
+		currentRoute,
+		session,
+		skipAssignedUsersData: !!skipAssignedUsersData,
+		userHasUpdateCasePermission,
+		caseOfficerUser,
+		inspectorUser
 	};
+};
 
-	Object.entries(submappers).forEach(([key, submapper]) => {
-		mappedData.appeal[key] = submapper({
-			appealDetails,
-			currentRoute,
-			session,
-			skipAssignedUsersData,
-			userHasUpdateCasePermission,
-			caseOfficerUser,
-			inspectorUser
-		});
-	});
+/**
+ * @param {InnerParams} params
+ * @returns {string}
+ */
+const getAppealType = ({ appealDetails }) => {
+	if (appealDetails === undefined) {
+		throw new Error('appealDetails is undefined');
+	}
+
+	if (!appealDetails.appealType) {
+		throw new Error('No appealType on appealDetails');
+	}
+	return appealDetails.appealType;
+};
 
-	return mappedData;
-}
+export const initialiseAndMapAppealData = initialiseAndMapDataFactory(
+	getSubmapperParams,
+	submaps,
+	getAppealType,
```

Comment:
As above, it seems a little overcomplicated to pass a function here when I think you could just pass the params themseves

Reply:
We haven't worked out the appeal type when we call the mappers so that's not really possible. Also bear in mind with these outer params you don't need to supply then when you actually use the mapper, they're only important inside this file. We'll only need to access this file to add new appeal types, and that's just adding a new line to the submapper dictionary